pkg/certificate/providers/certmanager/certificate_manager.go

```diff
@@ -82,7 +82,10 @@ func (cm *CertManager) getFromCache(cn certificate.CommonName) certificate.Certi
 func (cm *CertManager) RotateCertificate(cn certificate.CommonName) (certificate.Certificater, error) {
 	start := time.Now()
 
-	newCert, err := cm.issue(cn, cm.cfg.GetServiceCertValidityPeriod())
+	if cm.validityPeriod == 0 {
+		cm.validityPeriod = cm.cfg.GetServiceCertValidityPeriod()
+	}
```

Comment:
I'd leave a comment here why this is being done this way, otherwise it could be confusing.